app/models/user.rb

```diff
@@ -51,16 +52,18 @@ def self.search(term, role)
   end
 
   # notifications
-  def notify(title, message, link = '')
-    notify_slack("*#{title}*\n#{message}\n#{link}")
+  def notify(title, message, link = '', type: :default)
+    last_notification = notifications.first
+    # Set the `created_at` attribute, so that it can be compared by the `duplicate_of`
+    notification = Notification.new title: title, message: message, notification_type: type, user_id: id, read: false, link: link, created_at: DateTime.current
 
-    NotificationMailer.with(user: self, title: '[HART] ' + title.to_s, message: (message + link).to_s).notify_email.deliver_now if email_notifications
-
-    notification = Notification.new title: title, message: message
-    notification.user_id = id
-    notification.read = false
-    notification.link = link
-    notification.save # saving might fail, but there is no useful way to handle the error.
+    if notification.duplicate_of last_notification
+      last_notification.update(count: last_notification.count + 1)
+    else
+      notify_slack("*#{title}*\n#{message}\n#{link}")
+      NotificationMailer.with(user: self, title: '[HART] ' + title.to_s, message: (message + link).to_s).notify_email.deliver_now if email_notifications
+      notification.save # saving might fail, but there is no useful way to handle the error.
```

Comment:
> but there is no useful way to handle the error.
Well... we do have this `HartFormatter` that can log errors, which is introduced in this PR...

Reply:
That would not make sense, when saving the notification fails, we would log an error which would create a new Notification, which would most likely also fail to save, which would cause another error log...

We could write this in the comment however

Reply:
Oh, good point! I actually misread that line as somehow having to do with the `NotificationMailer` failing...
Thanks for adding the comment, that clears things up nicely.